app/bin/tools/remove_package.dart

```diff
@@ -161,13 +161,28 @@ Future removePackageVersion(String packageName, String version) async {
 
     final versionsQuery = T.query<PackageVersion>(packageKey);
     final versions = await versionsQuery.run().toList();
-    final versionNames = versions.map((v) => '${v.semanticVersion}').toList();
+    final versionNames = versions.map((v) => v.version).toList();
     if (!versionNames.contains(version)) {
       print('Package $packageName does not have a version $version.');
     }
 
-    if ('${package.latestSemanticVersion}' == version) {
-      throw Exception('Cannot delete the latest version of $packageName.');
+    if (versionNames.length == 1 && versionNames.single == version) {
+      throw Exception(
+          'Last version detected. Use full package removal without the version qualifier.');
+    }
+
+    bool updatePackage = false;
+    if (package.latestVersion == version) {
+      package.latestVersionKey = null;
+      updatePackage = true;
+    }
+    if (package.latestDevVersion == version) {
+      package.latestDevVersionKey = null;
```

Comment:
Don't we need to set this to something?

Reply:
Yeah, it looks like we always populate this, even if a package has no dev version. I'd keep it this way now, and we can do a cleanup of that later.